src/core/utils/layerutils.cpp

```diff
@@ -203,3 +206,44 @@ bool LayerUtils::deleteFeature( QgsProject *project, QgsVectorLayer *layer, cons
 
   return isSuccess;
 }
+
+QgsFeature LayerUtils::duplicateFeature( QgsVectorLayer *layer, const QgsFeature &feature )
+{
+  if ( !layer )
+  {
+    QgsMessageLog::logMessage( tr( "Cannot start editing, no layer" ), "QField", Qgis::Warning );
+    return QgsFeature();
+  }
+
+  if ( !layer->startEditing() || !layer->editBuffer() )
+  {
+    QgsMessageLog::logMessage( tr( "Cannot start editing" ), "QField", Qgis::Warning );
+    return QgsFeature();
+  }
+
```

Comment:
```suggestion
  if ( !feature.isValid() )
  {
    QgsMessageLog::logMessage( tr( "Cannot copy invalid feature" ), "QField", Qgis::Warning );
    return QgsFeature();
  }
```
Does it make sense to check the feature validity too?